WWST-1019 Add IKEA TRADFRI RGB bulbs

dkirker opened the pull request:
This is a work in progress. There are still a few rough edges as this is the first implementation of CIE 1931 xyY color.

dkirker committed:
WWST-1019 Add IKEA TRADFRI RGB bulbs

varzac commented on devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy:
```diff
@@ -73,14 +98,33 @@ def parse(String description) {
+			}
+
+			if (!shouldUseHueSaturation() && state.currentX && state.currentY) {
+				def hsv = safeColorXy2Hsv(state.currentX, state.currentY)
```
Did you do testing of this on the cloud side.  I was working on this locally and the hue/sat values that we get out don't really match up with the colors.

dkirker committed:
Code cleanup
Played around with sRGB(D65) profile from http://brucelindbloom.com/index.html?Eqn_RGB_XYZ_Matrix.html

dkirker committed:
Comment out most of the x,y conversion logging

dkirker committed:
Code cleanup and bring in changes from local execution implementation

varzac commented:
Tom and I were talking and we've decided that we are OK with just updating the min hubcore version to 25 to handle the issue of people joining on pre 25 hubs.

dkirker commented:
Thanks! I will update this PR.

dkirker committed:
Update min hubcore version to 0.25

dkirker committed:
Update text in copyright notice

tpmanley commented on devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy:
```diff
@@ -1,5 +1,5 @@
 /**
- *  Copyright 2017 SmartThings
+ *  Copyright 2017-2019 SmartThings
```
Best not to put an end date. Without it the copyright continues to the current date (I think)

dkirker replied:
Hmmmm Maybe that is what I had heard. Though, a quick look seems to indicate that really none of it matters? But, I'll revert that commit, and fix the couple of other code formatting things that I missed.

tpmanley commented on devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy:
```diff
@@ -67,27 +83,46 @@ def parse(String description) {
 				sendEvent(name: "saturation", value: saturationValue, displayed:false)
+			} else if(zigbeeMap.attrInt == ATTRIBUTE_X) { //X Attribute
+				state.currentRawX = zigbee.convertHexToInt(zigbeeMap.value)
+				//log.debug "xValue = $state.currentRawX"
```
Should remove commented out code

tpmanley commented on devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy:
```diff
@@ -67,27 +83,46 @@ def parse(String description) {
 
 		if (zigbeeMap?.clusterInt == COLOR_CONTROL_CLUSTER) {
-			if(zigbeeMap.attrInt == ATTRIBUTE_HUE){  //Hue Attribute
+			if(zigbeeMap.attrInt == ATTRIBUTE_HUE && shouldUseHueSaturation()){ //Hue Attribute
```
Nits: space after ifs

tpmanley commented on devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy:
```diff
@@ -67,27 +83,46 @@ def parse(String description) {
 			}
-		}
-		else {
+
```
Nit: extra line

dkirker replied:
I think diff view made this look weird. :/

tpmanley reviewed (dismissed):
Just a few minor comments

dkirker committed:
Revert "Update text in copyright notice"

This reverts commit 638e1aec0b1cd5e58ba831e586b19f3a49208fd3.

dkirker committed:
Formatting cleanup

dkirker committed:
Comment cleanup
